src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoder.cs

```diff
@@ -87,21 +88,30 @@ public override unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf
                 int idx = 0;
 
 #if NETCOREAPP
-                if (Sse2.IsSupported)
+                if (Sse2.IsSupported || AdvSimd.Arm64.IsSupported)
                 {
                     sbyte* startingAddress = (sbyte*)ptr;
                     while (utf8Text.Length - 16 >= idx)
                     {
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
-                        // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        bool containsNonAsciiBytes;
 
-                        // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
-                        // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        // Load the next 16 bytes, and check for ASCII text.
+                        // Any byte that's not in the ASCII range will already be negative when casted to signed byte.
+                        if (Sse2.IsSupported)
+                        {
+                            Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = Sse2Helper.ContainsNonAsciiByte(sourceValue);
+                        }
+                        else
+                        {
+                            Debug.Assert(AdvSimd.Arm64.IsSupported);
```

Comment:
>Debug.Assert(AdvSimd.Arm64.IsSupported); [](start = 28, length = 40)

You already have an assert inside `ContainsNonAsciiByte()`. So this is not needed.

Reply:
See my other comment on these asserts.